File: src/main/java/grp/faq/dao/AnswerMapper.java
package grp.faq.dao;

import grp.faq.entity.Answer;

public interface AnswerMapper {
    int deleteByPrimaryKey(Long cAnsid);

    int insert(Answer record);

    int insertSelective(Answer record);

    Answer selectByPrimaryKey(Long cAnsid);

    int updateByPrimaryKeySelective(Answer record);

    int updateByPrimaryKey(Answer record);

    int count();
}

File: src/main/scala/grp/faq/dao/AnswerMapper.scala
package grp.faq.dao

import java.util

import grp.faq.entity.Answer
import org.apache.ibatis.annotations.Param
import java.util.List
import java.util.Map

trait AnswerMapper {
  def getAnswerByUsername(@Param("username")username: String, @Param("index")index: Int): List[Map[String, String]]

  def update(answer: Answer): Int

  def delete(@Param("ansid")ansid: Int): Int

  def addAnswer(answer: Answer): Int

  def getAnswer(@Param("qid")qid: Int, @Param("uid")uid: Int): List[Map[Object,Object]]

  def getTodayAnswer(): Long
}